docs-src/content/functions/crypto.yml

```
@@ -103,6 +103,52 @@ funcs:
       - |
         $ gomplate -i '{{ "hello world" | crypto.EncryptAES "swordfish" 128 | base64.Encode }}'
         MnRutHovsh/9JN3YrJtBVjZtI6xXZh33bCQS2iZ4SDI=
+  - name: crypto.ECDSAGenerateKey
+    experimental: true
+    description: |
+      Generate a new Elliptic Curve Private Key and output in
+      PEM-encoded PKCS#1 ASN.1 DER form.
+
+      Go's standard NIST P-224, P-256, P-384, and P-521 elliptic curves are all
+      supported.
+
+      Default curve is P-256 and can be overridden with the optional `curve`
+      parameter.
+    pipeline: true
+    arguments:
+      - name: curve
+        required: false
+        description: |
+          One of Go's standard NIST curves, P-224, P-256, P-384, or P-521 -
+          defaults to P-256.
+    examples:
+      - |
+        $ gomplate -i '{{ crypto.ECDSAGenerateKey }}'
+        -----BEGIN EC PRIVATE KEY-----
+        ...
+      - |
+        $ gomplate -i '{{ $key := crypto.ECDSAGenerateKey "P-521" -}}
+          {{ $pub := crypto.ECDSADerivePublicKey $key -}}'
+        hello
+  - name: crypto.ECDSADerivePublicKey
+    experimental: true
+    description: |
+      Derive a public key from an elliptic curve private key and output in PKIX
+      ASN.1 DER form.
+    pipeline: true
+    arguments:
+      - name: key
+        required: true
+        description: the private key to derive a public key from
+    examples:
+      - |
+        $ gomplate -i '{{ crypto.ECDSAGenerateKey | crypto.ECDSADerivePublicKey }}'
+        -----BEGIN PUBLIC KEY-----
+        ...
+      - |
+        $ gomplate -c privKey=./privKey.pem \
+          -i '{{ $pub := crypto.ECDSADerivePublicKey .privKey -}}'
+        hello
```

This'll fail with an error like:

```
20:40:00 ERR  error="Datasources of type application/x-x509-ca-cert not yet supported" experimental=true
```

We'll need to use `-d` instead to defer parsing, and then we can use `include` to include without parsing. Alternatively we could use `file.Read`, but I don't really like to encourage that 😉

How about this:

```suggestion
        $ gomplate -d key=priv.pem -i '{{ crypto.ECDSADerivePublicKey (include "key") }}'
        -----BEGIN PUBLIC KEY-----
        MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQBZvTS1wcCJSsGYQUVoSVctynkuhke
        kikB38iNwx/80jzdm+Z8OmRGlwH6OE9NX1MyxjvYMimhcj6zkaOKh1/HhMABrfuY
        +hIz6+EUt/Db51awO7iCuRly5L4TZ+CnMAsIbtUOqsqwSQDtv0AclAuogmCst75o
        aztsmrD79OXXnhUlURI=
        -----END PUBLIC KEY-----
```